File: go/internal/entity/playlist.go
package entity

import "time"

// Playlist est une liste ordonnée de tracks appartenant à un user.
// is_queue = true → file de lecture active.
type Playlist struct {
	ID         string    `json:"id"`
	OwnerID    string    `json:"owner_id"`
	Title      string    `json:"title"`
	IsQueue    bool      `json:"is_queue"`
	TrackCount int       `json:"track_count"` // maintenu par trigger BDD (migration 005)
	CreatedAt  time.Time `json:"created_at"`
	Tracks     []Track   `json:"tracks,omitempty"`
}

// Track est un fichier audio référencé dans une playlist.
// Le fichier audio est stocké dans Supabase Storage (bucket: audio).
type Track struct {
	ID         string    `json:"id"`
	Title      string    `json:"title"`
	Artist     string    `json:"artist"`
	Duration   int       `json:"duration"` // secondes
	FileURL    string    `json:"file_url"` // URL Supabase Storage
	UploadedBy string    `json:"uploaded_by"`
	CreatedAt  time.Time `json:"created_at"`

	// Champs de liaison playlist (remplis quand la track est dans une playlist)
	PlaylistID string    `json:"playlist_id,omitempty"`
	Position   int       `json:"position,omitempty"`
	AddedAt    time.Time `json:"added_at,omitempty"`
}

File: flutter/lib/domain/entities/playlist.dart
class Playlist {
  final String id;
  final String ownerId;
  final String title;
  final bool isQueue;
  final int trackCount; // maintenu par trigger BDD (migration 005)
  final List<Track> tracks;

  const Playlist({
    required this.id,
    required this.ownerId,
    required this.title,
    required this.isQueue,
    required this.trackCount,
    this.tracks = const [],
  });
}

/// Track = fichier audio stocké dans Supabase Storage (bucket: audio).
class Track {
  final String id;
  final String title;
  final String artist;
  final int duration; // secondes
  final String fileUrl; // URL Supabase Storage
  final String uploadedBy;

  // Champs de liaison playlist (présents quand la track est dans une playlist)
  final String? playlistId;
  final int? position;

  const Track({
    required this.id,
    required this.title,
    required this.artist,
    required this.duration,
    required this.fileUrl,
    required this.uploadedBy,
    this.playlistId,
    this.position,
  });
}
